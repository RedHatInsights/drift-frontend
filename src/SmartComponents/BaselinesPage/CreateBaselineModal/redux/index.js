export { default as createBaselineModalActions } from './actions';
