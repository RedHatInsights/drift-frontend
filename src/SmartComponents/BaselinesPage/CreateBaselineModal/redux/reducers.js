import types from './types';
import helpers from '../../../helpers';

const initialState = {
    createBaselineModalOpened: false,
    baselineDataLoading: false,
    baselineData: undefined,
    createBaselineError: {}
};

export function createBaselineModalReducer(state = initialState, action) {
    let response = '';
    let errorObject = {};

    switch (action.type) {
        case `${types.TOGGLE_CREATE_BASELINE_MODAL}`:
            return {
                ...state,
                createBaselineModalOpened: !state.createBaselineModalOpened,
                createBaselineError: {}
            };
        case `${types.CREATE_BASELINE}_PENDING`:
            return {
                ...state,
                baselineDataLoading: true,
                baselineData: {},
                createBaselineError: {}
            };
        case `${types.CREATE_BASELINE}_FULFILLED`:
            return {
                ...state,
                baselineDataLoading: false,
                baselineData: action.payload
            };
        case `${types.CREATE_BASELINE}_REJECTED`:
            response = action.payload.response;

            if (response.data === '') {
                errorObject = { detail: helpers.makeSentenceCase(response.statusText), status: response.status };
            } else if (response.data.message) {
                errorObject = { detail: helpers.makeSentenceCase(response.data.message), status: response.status };
            } else {
                errorObject = { detail: helpers.makeSentenceCase(response.data.detail), status: response.status };
            }

            return {
                ...state,
                baselineDataLoading: false,
                createBaselineError: errorObject
            };

        default:
            return state;
    }
}
