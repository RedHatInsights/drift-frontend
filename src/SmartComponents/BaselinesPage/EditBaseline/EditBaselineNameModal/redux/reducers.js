import types from './types';

const initialState = {
    editNameModalOpened: false
};

export function editNameModalReducer(state = initialState, action) {
    switch (action.type) {
        case `${types.TOGGLE_EDIT_NAME_MODAL}`:
            return {
                ...state,
                editNameModalOpened: !state.editNameModalOpened
            };

        default:
            return state;
    }
}
