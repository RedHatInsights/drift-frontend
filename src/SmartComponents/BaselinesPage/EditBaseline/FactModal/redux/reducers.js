import types from './types';

const initialState = {
    factModalOpened: false,
    factName: '',
    factValue: '',
    factData: []
};

export function factModalReducer(state = initialState, action) {
    switch (action.type) {
        case `${types.TOGGLE_FACT_MODAL}`:
            return {
                ...state,
                factModalOpened: !state.factModalOpened
            };
        case `${types.SET_FACT_DATA}`:
            return {
                ...state,
                factName: action.payload.factName,
                factValue: action.payload.factValue,
                factData: action.payload.fact
            };

        default:
            return state;
    }
}
