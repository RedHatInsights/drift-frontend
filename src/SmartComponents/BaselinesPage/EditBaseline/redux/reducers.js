import types from './types';
import editBaselineHelpers from '../helpers';
import { FACT_VALUE } from '../../../../constants';

const initialState = {
    baselineData: undefined,
    baselineDataLoading: false,
    editBaselineTableData: [],
    editNameModalOpened: false,
    error: {},
    expandedRows: [],
    factModalOpened: false,
    factName: '',
    factValue: '',
    factData: [],
    isCategory: false,
    isSubFact: false,
    selectAll: false
};

export function editBaselineReducer(state = initialState, action) {
    let errorObject = {};
    let newEditBaselineTableData = [];
    let newExpandedRows = [];
    let response;

    switch (action.type) {
        case `${types.TOGGLE_EDIT_NAME_MODAL}`:
            return {
                ...state,
                editNameModalOpened: !state.editNameModalOpened,
                error: {}
            };
        case `${types.FETCH_BASELINE_DATA}_PENDING`:
            return {
                ...state,
                baselineDataLoading: true
            };
        case `${types.FETCH_BASELINE_DATA}_FULFILLED`:
            newEditBaselineTableData = editBaselineHelpers.buildBaselineTableData(action.payload.baseline_facts);
            return {
                ...state,
                baselineDataLoading: false,
                baselineData: action.payload,
                editBaselineTableData: newEditBaselineTableData,
                selectAll: false
            };
        case `${types.PATCH_BASELINE}_PENDING`:
            return {
                ...state,
                baselineDataLoading: true
            };
        case `${types.PATCH_BASELINE}_FULFILLED`:
            return {
                ...state,
                baselineDataLoading: false,
                baselineData: action.payload
            };
        case `${types.PATCH_BASELINE}_REJECTED`:
            response = action.payload.response;

            if (response.data === '') {
                errorObject = { detail: response.statusText, status: response.status };
            } else if (response.data.message) {
                errorObject = { detail: response.data.message, status: response.status };
            } else {
                errorObject = { detail: response.data.detail, status: response.status };
            }

            return {
                ...state,
                baselineDataLoading: false,
                error: errorObject
            };
        case `${types.EXPAND_PARENT_FACT}`:
            newExpandedRows = editBaselineHelpers.toggleExpandedRow(state.expandedRows, action.payload);
            return {
                ...state,
                expandedRows: newExpandedRows
            };
        case `${types.TOGGLE_FACT_MODAL}`:
            return {
                ...state,
                error: {},
                factModalOpened: !state.factModalOpened
            };
        case `${types.SET_FACT_DATA}`:
            return {
                ...state,
                factName: action.payload.factName,
                factValue: action.payload.factValue,
                factData: action.payload.fact,
                isCategory: action.payload.isCategory,
                isSubFact: action.payload.isSubFact
            };
        case `${types.SELECT_FACT}`: {
            newEditBaselineTableData = [ ...state.editBaselineTableData ];

            newEditBaselineTableData.map(row => {
                let factId = row[0];
                if (action.payload.ids.includes(factId)) {
                    row.selected = action.payload.isSelected;
                }

                if (editBaselineHelpers.isCategory(row) && row[FACT_VALUE].length > 0) {
                    editBaselineHelpers.baselineSubFacts(row).map(subFact => {
                        let subFactId = subFact[0];
                        if (action.payload.ids.includes(subFactId)) {
                            subFact.selected = action.payload.isSelected;
                        }
                    });

                    row.selected = editBaselineHelpers.isAllSelected(editBaselineHelpers.baselineSubFacts(row));
                }
            });

            let isAllSelected = editBaselineHelpers.isAllSelected(newEditBaselineTableData);

            return {
                ...state,
                editBaselineTableData: newEditBaselineTableData.slice(),
                selectAll: isAllSelected
            };
        }

        case `${types.CLEAR_BASELINE_DATA}`:
            return {
                ...state,
                baselineData: undefined
            };

        default:
            return state;
    }
}
