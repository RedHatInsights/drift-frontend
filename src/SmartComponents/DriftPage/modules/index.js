export { default as DriftPageActions } from './actions';
export { default as DriftPageTypes } from './types';
export { default as DriftPageReducers } from './reducers';
