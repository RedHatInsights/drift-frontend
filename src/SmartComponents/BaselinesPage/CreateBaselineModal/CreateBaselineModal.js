import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { useDispatch, useSelector } from 'react-redux';
import { Button, Modal, Radio, TextInput, Form, FormGroup, ValidatedOptions } from '@patternfly/react-core';
import { sortable, cellWidth } from '@patternfly/react-table';
import { addNewListener } from '../../../store';

import SystemsTable from '../../SystemsTable/SystemsTable';
import BaselinesTable from '../../BaselinesTable/BaselinesTable';
import GlobalFilterAlert from '../../GlobalFilterAlert/GlobalFilterAlert';
import { createBaselineModalActions } from './redux';
import { baselinesTableActions } from '../../BaselinesTable/redux';
import systemsTableActions from '../../SystemsTable/actions';
import useInsightsNavigate from '@redhat-cloud-services/frontend-components-utilities/useInsightsNavigate';

const CreateBaselineModal = ({ middlewareListener, permissions }) => {
    const dispatch = useDispatch();
    const navigate = useInsightsNavigate();
    const [ baselineName, setBaselineName ] = useState('');
    const [ fromScratchChecked, setFromScratchChecked ] = useState(true);
    const [ copyBaselineChecked, setCopyBaselineChecked ] = useState(false);
    const [ copySystemChecked, setCopySystemChecked ] = useState(false);
    const columns = [
        { title: 'Name', transforms: [ sortable, cellWidth(60) ]},
        { title: 'Last updated', transforms: [ sortable, cellWidth(20) ]},
        { title: 'Associated systems', transforms: [ cellWidth(20) ]}
    ];

    const createBaselineModalOpened = useSelector(({ createBaselineModalState }) => createBaselineModalState.createBaselineModalOpened);
    const baselineData = useSelector(({ createBaselineModalState }) => createBaselineModalState.baselineData);
    const entities = useSelector(({ entities }) => entities);
    const selectedBaselineIds = useSelector(({ baselinesTableState }) => baselinesTableState.radioTable.selectedBaselineIds);
    const createBaselineError = useSelector(({ createBaselineModalState }) => createBaselineModalState.createBaselineError);
    const loading = useSelector(({ baselinesTableState }) => baselinesTableState.radioTable.loading);
    const emptyState = useSelector(({ baselinesTableState }) => baselinesTableState.radioTable.emptyState);
    const baselineTableData = useSelector(({ baselinesTableState }) => baselinesTableState.radioTable.baselineTableData);
    const totalBaselines = useSelector(({ baselinesTableState }) => baselinesTableState.radioTable.totalBaselines);
    const globalFilterState = useSelector(({ globalFilterState }) => globalFilterState);
    const baselineError = useSelector(({ baselinesTableState }) => baselinesTableState.radioTable.baselineError);

    const deselectHistoricalProfiles = async () => {
        await dispatch(systemsTableActions.updateColumns('display_name'));
        dispatch(systemsTableActions.selectSingleHSP());
    };

    useEffect(() => {
        if (middlewareListener) {
            window.entityListener = addNewListener(middlewareListener, {
                actionType: 'SELECT_ENTITY',
                callback: () => {
                    createBaselineModalOpened ? deselectHistoricalProfiles() : null;
                }
            });

            window.entityListener = addNewListener(middlewareListener, {
                actionType: 'SELECT_SINGLE_HSP',
                callback: () => {
                    dispatch(systemsTableActions.updateColumns('display_selected_hsp'));
                }
            });
        }

        return () => {
            window.removeEventListener('SELECT_ENTITY', deselectHistoricalProfiles);
            window.removeEventListener('SELECT_SINGLE_HSP', systemsTableActions.updateColumns);
        };
    }, []);

    useEffect(() => {
        if (baselineData?.id) {
            navigate(`/baselines/${baselineData.id}`);
        }
    }, [ baselineData ]);

    const cancelModal = () => {
        setBaselineName('');
        dispatch(baselinesTableActions.clearSelectedBaselines('RADIO'));
        dispatch(systemsTableActions.selectSingleHSP());
        dispatch(createBaselineModalActions.toggleCreateBaselineModal());
    };

    const onSelect = (event, isSelected, rowId) => {
        let id = [ baselineTableData[rowId][0] ];
        dispatch(baselinesTableActions.selectBaseline(id, isSelected, 'RADIO'));
    };

    const findSelectedRadio = () => {
        const radioChecked = { copyBaselineChecked, copySystemChecked, fromScratchChecked };
        let keys = Object.keys(radioChecked);
        let selectedKey;

        keys.forEach(function(key) {
            if (radioChecked[key]) {
                selectedKey = key.substring(0, key.length - 7).toLowerCase();
            }
        });

        return selectedKey;
    };

    const submitBaselineName = async () => {
        /*eslint-disable camelcase*/
        let newBaselineObject = { display_name: baselineName };

        try {
            if (baselineName !== '') {
                if (fromScratchChecked) {
                    newBaselineObject.baseline_facts = [];
                    await dispatch(createBaselineModalActions.createBaseline(newBaselineObject));
                } else if (selectedBaselineIds.length === 1 && copyBaselineChecked) {
                    newBaselineObject = { display_name: baselineName };
                    await dispatch(createBaselineModalActions.createBaseline(newBaselineObject, selectedBaselineIds[0]));
                } else if (entities?.selectedSystemIds.length && copySystemChecked) {
                    newBaselineObject.inventory_uuid = entities?.selectedSystemIds[0];
                    await dispatch(createBaselineModalActions.createBaseline(newBaselineObject));
                } else if (entities?.selectedHSP && copySystemChecked) {
                    newBaselineObject.hsp_uuid = entities.selectedHSP.id;
                    await dispatch(createBaselineModalActions.createBaseline(newBaselineObject));
                }

                dispatch(createBaselineModalActions.toggleCreateBaselineModal());
                dispatch(createBaselineModalActions.selectSingleHSP());
                dispatch(baselinesTableActions.clearSelectedBaselines('RADIO'));
            }
        } catch (e) {
            // do nothing and let redux handle
        }
        /*eslint-enable camelcase*/
    };

    const handleChecked = (_, event) => {
        const value = event.currentTarget.value;
        dispatch(baselinesTableActions.clearSelectedBaselines('RADIO'));

        if (value === 'fromScratchChecked') {
            setFromScratchChecked(true);
            setCopyBaselineChecked(false);
            setCopySystemChecked(false);
        } else if (value === 'copyBaselineChecked') {
            setFromScratchChecked(false);
            setCopyBaselineChecked(true);
            setCopySystemChecked(false);
        } else {
            setFromScratchChecked(false);
            setCopyBaselineChecked(false);
            setCopySystemChecked(true);
        }
    };

    const renderRadioButtons = () => {
        return (<React.Fragment>
            <Radio
                isChecked={ fromScratchChecked }
                id='create-baseline'
                data-testid='from-scratch-radio'
                ouiaId='create-baseline-from-scratch-radio'
                name='baseline-create-options'
                label='Create baseline from scratch'
                value='fromScratchChecked'
                onChange={ handleChecked }
            />
            <Radio
                isChecked={ copyBaselineChecked }
                id='copy-baseline'
                ouiaId='create-baseline-copy-baseline-radio'
                data-testid='copy-baseline-radio'
                name='baseline-create-options'
                label='Copy an existing baseline'
                value='copyBaselineChecked'
                onChange={ handleChecked }
            />
            <Radio
                isChecked={ copySystemChecked }
                id='copy-system'
                ouiaId='create-baseline-copy-system-radio'
                data-testid='copy-system-radio'
                name='baseline-create-options'
                label='Copy an existing system or historical profile'
                value='copySystemChecked'
                onChange={ handleChecked }
            />
        </React.Fragment>
        );
    };

    const renderActions = () => {
        let actions;

        if (baselineName === ''
            || (copyBaselineChecked && selectedBaselineIds.length === 0)
            || (copySystemChecked &&
                (!entities?.selectedSystemIds.length && !entities?.selectedHSP)
            )
        ) {
            actions = [
                <Button
                    key="confirm"
                    variant="primary"
                    isDisabled
                    ouiaId={ `create-baseline-button-${findSelectedRadio()}` }
                    aria-label='create-baseline-disabled'
                >
                    Create baseline
                </Button>,
                <Button
                    key="cancel"
                    variant="link"
                    onClick={ () => cancelModal() }
                    ouiaId="create-baseline-modal-cancel-button"
                    aria-label='cancel-create-baseline'
                >
                    Cancel
                </Button>
            ];
        } else {
            actions = [
                <Button
                    key="confirm"
                    variant="primary"
                    onClick={ () => submitBaselineName() }
                    ouiaId={ `create-baseline-button-${findSelectedRadio()}` }
                    aria-label='create-baseline-confirm'
                >
                    Create baseline
                </Button>,
                <Button
                    key="cancel"
                    variant="link"
                    onClick={ () => cancelModal() }
                    ouiaId="create-baseline-modal-cancel-button"
                    aria-label='cancel-create-baseline'
                >
                    Cancel
                </Button>
            ];
        }

        return actions;
    };

    const buildSystemColumns = (originalColumns) => {
        let columns = originalColumns.map(function(column) {
            if (column.key === 'display_name' || column.key === 'display_selected_hsp') {
                return {
                    title: 'Name',
                    key: column.key,
                    props: { width: 20 },
                    renderFunc: (value, _id, { display_selected_hsp: selectedHSP }) => {
                        return selectedHSP || value;
                    }
                };
            } else {
                return column;
            }
        });

        if (permissions.hspRead) {
            columns.push({ key: 'historical_profiles', props: { width: 10, isStatic: true }, title: 'Historical profiles' });
        }

        return columns;
    };

    const renderCopySystem = () => {
        return (<React.Fragment>
            <b>Select system or historical profile to copy from</b>
            <br></br>
            <SystemsTable
                createBaselineModal={ true }
                hasMultiSelect={ false }
                historicalProfiles={ entities?.selectedHSP ? [ entities.selectedHSP ] : [] }
                permissions={ permissions }
                entities={ entities }
                selectVariant='radio'
                systemColumns={ buildSystemColumns }
                deselectHistoricalProfiles={ deselectHistoricalProfiles }
            />
        </React.Fragment>
        );
    };

    const renderCopyBaseline = () => {
        return (<React.Fragment>
            <b>Select baseline to copy from</b>
            <BaselinesTable
                tableId='RADIO'
                onSelect={ onSelect }
                tableData={ baselineTableData }
                loading={ loading }
                columns={ columns }
                totalBaselines={ totalBaselines }
                permissions={ permissions }
                hasMultiSelect={ false }
                selectedBaselineIds={ selectedBaselineIds }
                leftAlignToolbar={ true }
                hasSwitch={ false }
                revertBaselineFetch={ baselinesTableActions.revertBaselineFetch }
                baselineError={ baselineError }
                emptyState={ emptyState }
            />
        </React.Fragment>
        );
    };

    const checkKeyPress = (event) => {
        if (event.key === 'Enter') {
            event.preventDefault();
            baselineName ? submitBaselineName() : null;
        }
    };

    const renderModalBody = () => {
        let modalBody;

        if (copyBaselineChecked) {
            modalBody = renderCopyBaseline();
        } else if (copySystemChecked) {
            modalBody = renderCopySystem();
        }

        return (<React.Fragment>
            { renderRadioButtons() }
            <div className='md-padding-top md-padding-bottom'>
                <Form>
                    <FormGroup
                        label='Baseline name'
                        isRequired
                        type="text"
                        helperTextInvalid={ Object.prototype.hasOwnProperty.call(createBaselineError, 'detail') ? createBaselineError.detail : null }
                        fieldId="name"
                        validated={ Object.prototype.hasOwnProperty.call(createBaselineError, 'status') ? 'error' : null }
                        onKeyPress={ checkKeyPress }
                    >
                        <TextInput
                            className="fact-value"
                            value={ baselineName }
                            type="text"
                            onChange={ (value) => setBaselineName(value) }
                            validated={ Object.prototype.hasOwnProperty.call(createBaselineError, 'status') ? ValidatedOptions.error : null }
                            aria-label="baseline name"
                        />
                    </FormGroup>
                </Form>
            </div>
            { modalBody }
        </React.Fragment>
        );
    };

    return (
        <Modal
            className="drift create-baseline-modal"
            aria-label='create-baseline-modal'
            width="1200px"
            title="Create baseline"
            isOpen={ createBaselineModalOpened }
            onClose={ cancelModal }
            actions={ renderActions() }
        >
            { copySystemChecked
                ? <GlobalFilterAlert globalFilterState={ globalFilterState }/>
                : null
            }
            { renderModalBody() }
        </Modal>
    );
};

CreateBaselineModal.propTypes = {
    middlewareListener: PropTypes.object,
    permissions: PropTypes.object
};

export default CreateBaselineModal;
