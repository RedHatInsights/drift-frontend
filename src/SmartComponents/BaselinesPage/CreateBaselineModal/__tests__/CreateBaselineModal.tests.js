import React from 'react';
import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Provider } from 'react-redux';
import { createMiddlewareListener, init } from '../../../../store';
import api from '../../../../api';
import { globalFilterInitialState } from '../../../modules/reducers';
import tableDataFixtures from '../../../BaselinesTable/redux/__tests__/baselinesTableReducer.fixtures';
import CreateBaselineModal from '../CreateBaselineModal';

jest.mock('../../../../api');

const middlewareListener = createMiddlewareListener();
middlewareListener.getMiddleware();

describe('CreateBaselineModal', () => {
    let props;
    const store = init().registry.getStore();

    beforeEach(() => {
        props = {
            permissions: {
                hspRead: true,
                baselinesWrite: true,
                baselinesRead: true
            },
            middlewareListener
        };
        store.getState().globalFilterState = globalFilterInitialState;
        store.getState().createBaselineModalState.createBaselineModalOpened = true;
    });

    afterEach(() => {
        jest.clearAllMocks();
    });

    it('should render correctly', async () => {
        render(
            <Provider store={ store }>
                <CreateBaselineModal { ...props }/>
            </Provider>
        );
        const modal = screen.getByLabelText('create-baseline-modal');
        expect(modal).toMatchSnapshot();
    });

    it('should render error correctly', async() => {
        api.postNewBaseline.mockImplementation(async () => {
            return await Promise.reject({
                response: { data: {}, status: 404, detail: 'This is an error' }
            });
        });

        render(
            <Provider store={ store }>
                <CreateBaselineModal { ...props } />
            </Provider>
        );
        const input = screen.getByLabelText('baseline name');
        await waitFor(() => fireEvent.change(input, { target: { value: 'bogus' }}));
        await waitFor(() => userEvent.click(screen.getByLabelText('create-baseline-confirm')));

        expect(input).toBeInvalid();
    });

    it('should use handleChecked', async () => {
        render(
            <Provider store={ store }>
                <CreateBaselineModal { ...props } />
            </Provider>
        );

        const copyBaselineRadio = screen.getByTestId('copy-baseline-radio');
        await waitFor(() => userEvent.click(copyBaselineRadio));
        expect(copyBaselineRadio.checked).toEqual(true);

        const copySystemRadio = screen.getByTestId('copy-system-radio');
        await waitFor(() => userEvent.click(copySystemRadio));
        expect(copyBaselineRadio.checked).toEqual(false);
        expect(copySystemRadio.checked).toEqual(true);

        const fromScratchRadio = screen.getByTestId('from-scratch-radio');
        await waitFor(() => userEvent.click(fromScratchRadio));
        expect(copySystemRadio.checked).toEqual(false);
        expect(fromScratchRadio.checked).toEqual(true);
    });

    it('should update baselineName', async () => {
        render(
            <Provider store={ store }>
                <CreateBaselineModal { ...props }/>
            </Provider>
        );

        const input = screen.getByLabelText('baseline name');
        await waitFor(() => fireEvent.change(input, { target: { value: 'baselines rock' }}));
        expect(input.value).toEqual('baselines rock');
    });

    it('should submit baselineName with copyBaselineChecked', async () => {
        api.getBaselineList.mockImplementation(async () => {
            return tableDataFixtures.baselinesListPayload;
        });
        render(
            <Provider store={ store }>
                <CreateBaselineModal { ...props }/>
            </Provider>
        );

        const copyBaselineRadio = screen.getByTestId('copy-baseline-radio');
        await waitFor(() => userEvent.click(copyBaselineRadio));

        const input = screen.getByLabelText('baseline name');
        await waitFor(() => fireEvent.change(input, { target: { value: 'baselines rock' }}));

        await waitFor(() => userEvent.click(screen.getByLabelText('Select row 0')));

        await waitFor(() => userEvent.click(screen.getByLabelText('create-baseline-confirm')));

        expect(api.getBaselineList).toHaveBeenCalledTimes(1);
    });

    it('should cancelModal', async () => {
        api.getBaselineList.mockImplementation(async () => {
            return tableDataFixtures.baselinesListPayload;
        });

        render(
            <Provider store={ store }>
                <CreateBaselineModal { ...props }/>
            </Provider>
        );

        const copyBaselineRadio = screen.getByTestId('copy-baseline-radio');
        await waitFor(() => userEvent.click(copyBaselineRadio));

        await waitFor(() => userEvent.click(screen.getByLabelText('Select row 0')));
        expect(store.getState().baselinesTableState.radioTable.selectedBaselineIds).toEqual([ '1234' ]);

        const cancelButton = screen.getByLabelText('cancel-create-baseline');
        await waitFor(() => userEvent.click(cancelButton));
        expect(store.getState().baselinesTableState.radioTable.selectedBaselineIds).toEqual([]);
        expect(store.getState().createBaselineModalState.createBaselineModalOpened).toEqual(false);
    });
});
