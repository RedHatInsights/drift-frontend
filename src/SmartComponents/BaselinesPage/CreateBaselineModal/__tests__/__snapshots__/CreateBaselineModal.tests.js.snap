// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateBaselineModal should render correctly 1`] = `
<div
  aria-describedby="pf-modal-part-2"
  aria-label="create-baseline-modal"
  aria-labelledby="pf-modal-part-0 pf-modal-part-1"
  aria-modal="true"
  class="pf-c-modal-box drift create-baseline-modal"
  data-ouia-component-id="OUIA-Generated-Modal-default-1"
  data-ouia-component-type="PF4/ModalContent"
  data-ouia-safe="true"
  id="pf-modal-part-0"
  role="dialog"
  style="width: 1200px;"
>
  <button
    aria-disabled="false"
    aria-label="Close"
    class="pf-c-button pf-m-plain"
    data-ouia-component-id="OUIA-Generated-Modal-default-1-ModalBoxCloseButton"
    data-ouia-component-type="PF4/Button"
    data-ouia-safe="true"
    type="button"
  >
    <svg
      aria-hidden="true"
      fill="currentColor"
      height="1em"
      role="img"
      style="vertical-align: -0.125em;"
      viewBox="0 0 352 512"
      width="1em"
    >
      <path
        d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
      />
    </svg>
  </button>
  <header
    class="pf-c-modal-box__header"
  >
    <h1
      class="pf-c-modal-box__title"
      id="pf-modal-part-1"
    >
      
      <span
        class="pf-c-modal-box__title-text"
      >
        Create baseline
      </span>
    </h1>
  </header>
  <div
    class="pf-c-modal-box__body"
    id="pf-modal-part-2"
  >
    <div
      class="pf-c-radio"
    >
      <input
        aria-invalid="false"
        checked=""
        class="pf-c-radio__input"
        data-ouia-component-id="create-baseline-from-scratch-radio"
        data-ouia-component-type="PF4/Radio"
        data-ouia-safe="true"
        data-testid="from-scratch-radio"
        id="create-baseline"
        name="baseline-create-options"
        type="radio"
        value="fromScratchChecked"
      />
      <label
        class="pf-c-radio__label"
        for="create-baseline"
      >
        Create baseline from scratch
      </label>
    </div>
    <div
      class="pf-c-radio"
    >
      <input
        aria-invalid="false"
        class="pf-c-radio__input"
        data-ouia-component-id="create-baseline-copy-baseline-radio"
        data-ouia-component-type="PF4/Radio"
        data-ouia-safe="true"
        data-testid="copy-baseline-radio"
        id="copy-baseline"
        name="baseline-create-options"
        type="radio"
        value="copyBaselineChecked"
      />
      <label
        class="pf-c-radio__label"
        for="copy-baseline"
      >
        Copy an existing baseline
      </label>
    </div>
    <div
      class="pf-c-radio"
    >
      <input
        aria-invalid="false"
        class="pf-c-radio__input"
        data-ouia-component-id="create-baseline-copy-system-radio"
        data-ouia-component-type="PF4/Radio"
        data-ouia-safe="true"
        data-testid="copy-system-radio"
        id="copy-system"
        name="baseline-create-options"
        type="radio"
        value="copySystemChecked"
      />
      <label
        class="pf-c-radio__label"
        for="copy-system"
      >
        Copy an existing system or historical profile
      </label>
    </div>
    <div
      class="md-padding-top md-padding-bottom"
    >
      <form
        class="pf-c-form"
        novalidate=""
      >
        <div
          class="pf-c-form__group"
          type="text"
        >
          <div
            class="pf-c-form__group-label"
          >
            <label
              class="pf-c-form__label"
              for="name"
            >
              <span
                class="pf-c-form__label-text"
              >
                Baseline name
              </span>
              <span
                aria-hidden="true"
                class="pf-c-form__label-required"
              >
                 
                *
              </span>
            </label>
             
          </div>
          <div
            class="pf-c-form__group-control"
          >
            <input
              aria-invalid="false"
              aria-label="baseline name"
              class="pf-c-form-control fact-value"
              data-ouia-component-id="OUIA-Generated-TextInputBase-1"
              data-ouia-component-type="PF4/TextInput"
              data-ouia-safe="true"
              type="text"
              value=""
            />
            
          </div>
        </div>
      </form>
    </div>
  </div>
  <footer
    class="pf-c-modal-box__footer"
  >
    <button
      aria-disabled="true"
      aria-label="create-baseline-disabled"
      class="pf-c-button pf-m-primary pf-m-disabled"
      data-ouia-component-id="create-baseline-button-fromscratch"
      data-ouia-component-type="PF4/Button"
      data-ouia-safe="true"
      disabled=""
      type="button"
    >
      Create baseline
    </button>
    <button
      aria-disabled="false"
      aria-label="cancel-create-baseline"
      class="pf-c-button pf-m-link"
      data-ouia-component-id="create-baseline-modal-cancel-button"
      data-ouia-component-type="PF4/Button"
      data-ouia-safe="true"
      type="button"
    >
      Cancel
    </button>
  </footer>
</div>
`;
